sklearn/linear_model/tests/test_logistic.py

```diff
@@ -89,6 +93,49 @@ def test_error():
         assert_raise_message(ValueError, msg, LR(max_iter="test").fit, X, Y1)
 
 
+def test_logistic_cv_mock_scorer():
+
+    class MockScorer(object):
+        def __init__(self):
+            self.calls = 0
+            self.scores = [0.1, 0.4, 0.8, 0.5]
+
+        def __call__(self, model, X, y, sample_weight=None):
+            score = self.scores[self.calls % len(self.scores)]
+            self.calls += 1
+            return score
+
+    mock_scorer = MockScorer()
+    Cs = [1, 2, 3, 4]
+    cv = 2
+
+    lr = LogisticRegressionCV(Cs=Cs, scoring=mock_scorer, cv=cv)
+    lr.fit(X, Y1)
+
+    # Cs[2] has the highest score (0.8) from MockScorer
+    assert lr.C_[0] == Cs[2]
+
+    # scorer called 8 times (cv*len(Cs))
+    assert mock_scorer.calls == cv * len(Cs)
+
+    # reset mock_scorer
+    mock_scorer.calls = 0
+    with pytest.warns(ChangedBehaviorWarning):
+        custom_score = lr.score(X, lr.predict(X))
```

Comment:
you're not asserting that it warns, right?

Reply:
The `pytest.warns` context manager asserts that the inner block *does* warn. In this context, a `ChangedBehaviorWarning` warning appears when `scoring` is not `None`. This assertion is used to test the following lines in the `LogisticRegressionCV.score` function:

https://github.com/thomasjpfan/scikit-learn/blob/21af4b596023d3b3d66aaae3b3df8952a66a9e6a/sklearn/linear_model/logistic.py#L1816-L1821

Reply:
thanks, haven't used those much.